src/Alpha_complex/include/gudhi/Incremental_delaunay.h

```diff
@@ -0,0 +1,137 @@
+ *   to name vertices, and a member function `void insert_simplex_and_subfaces(vrange)` which accepts a range
+ *   of `Vertex_handle` (representing a simplex).
+ * @tparam PointRange Forward range of `K::Point_d`.
+ *
+ * @param[in] k The geometric kernel.
+ * @param[out] complex The simplicial complex.
+ * @param[in] points Embedding of the points.
+ *
+ * Authors: Michael Kerber, Marc Glisse
+ */
+template<typename K, typename SimplicialComplex, typename PointRange>
+void construct_incremental_delaunay(K const&k, SimplicialComplex& complex, PointRange const& points) {
+  using TDS = CGAL::Triangulation_data_structure<
+    typename K::Dimension,
+    CGAL::Triangulation_vertex<K, typename SimplicialComplex::Vertex_handle>,
+    CGAL::Triangulation_ds_full_cell<void, CGAL::TDS_full_cell_mirror_storage_policy> >;
+  using Triangulation = CGAL::Delaunay_triangulation<K, TDS>;
+  auto pt_it = std::begin(points);
+  auto pt_end = std::end(points);
+  if(pt_it == pt_end) return;
+  int d = k.point_dimension_d_object()(*pt_it);
+  Triangulation tri(d);
+
+  // local variables pulled out for recycling
+  typedef std::vector<typename Triangulation::Full_cell_handle> Full_cell_h_vector;
+  Full_cell_h_vector cs;
+  std::vector<typename SimplicialComplex::Vertex_handle> splx;
+
+  for(int idx = 0; pt_it != pt_end; ++pt_it, ++idx) {
+    // Adapted from Delaunay_triangulation::insert(Point)
+    auto&& p = *pt_it;
+    typename Triangulation::Locate_type lt;
+    typename Triangulation::Face f(d);
+    typename Triangulation::Facet ft;
+    typename Triangulation::Full_cell_handle s = tri.locate(p, lt, f, ft);
+    // Adapted from Delaunay_triangulation::insert(Point,Locate_type,Face,Facet,Full_cell_handle)
+    if (lt == Triangulation::OUTSIDE_AFFINE_HULL) {
+      // nothing disappears
+      tri.insert_outside_affine_hull(p)->data() = idx;
+    }
+    else if (lt == Triangulation::ON_VERTEX) {
+      // nothing disappears
+    }
+    else {
+      if(tri.current_dimension() == 1) {
+	if(lt == Triangulation::OUTSIDE_CONVEX_HULL) {
+	  // only an infinite face disappears
+	} else {
+	  complex.insert_simplex_and_subfaces({s->vertex(0)->data(), s->vertex(1)->data(), idx});
+        }
+        typename Triangulation::Vertex_handle v = tri.tds().insert_in_full_cell(s);
+        v->set_point(p);
+        v->data() = idx;
+      } else { // main case
+        // Adapted from Delaunay_triangulation::insert_in_conflicting_cell(Point,Full_cell_handle)
+	/* Full_cell_h_vector cs; */ cs.clear();
+	std::back_insert_iterator<Full_cell_h_vector> out(cs);
+	typename Triangulation::Facet ft = tri.compute_conflict_zone(p, s, out);
+	for(auto c : cs) {
+	  /* std::vector<int> splx; */ splx.clear();
+	  for(int i = 0; i <= tri.current_dimension(); ++i) {
+	    // We could skip the infinite vertex and output a (new) simplex, but that's not so useful
+	    if(tri.is_infinite(c->vertex(i))) goto next_conflicting_cell;
```

Comment:
I am still not a big fan of `goto` and I would prefer a solution with a `boolean` and `continue`, but we already discussed about it, the remark is minor.

Reply:
```diff
@@ -88,15 +88,16 @@ void construct_incremental_delaunay(K const&k, SimplicialComplex& complex, Point
         typename Triangulation::Facet ft = tri.compute_conflict_zone(p, s, out);
         for(auto c : cs) {
           /* std::vector<int> splx; */ splx.clear();
+          bool interrupted = false;
           for(int i = 0; i <= tri.current_dimension(); ++i) {
             // We could skip the infinite vertex and output a (new) simplex, but that's not so useful
-            if(tri.is_infinite(c->vertex(i))) goto next_conflicting_cell;
+            if(tri.is_infinite(c->vertex(i))) { interrupted = true; break; }
             splx.push_back(c->vertex(i)->data());
           }
+          if (interrupted) continue;
           // We could put idx at the beginning of the vector once instead of reinserting it each time
           splx.push_back(idx);
           complex.insert_simplex_and_subfaces(splx);
-next_conflicting_cell:;
         }
         tri.insert_in_hole(p, cs.begin(), cs.end(), ft)->data() = idx;
       }
```
doesn't look too horrible so I might change this one. But note that `goto` is the official syntax to do what some would like to call `continue(2)` (`continue` but for a loop that is not the innermost one), and the usual `continue` is defined in terms of `goto` in the standard.

![image](https://github.com/GUDHI/gudhi-devel/assets/10097863/5ac111df-2ef1-4d64-9f98-c16c0e958d3a)